src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/Registration/RegistrationIdValidator.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    public class RegistrationIdValidator : RegistrationExistsValidator
+    {
+        public RegistrationIdValidator(
+            IDictionary<FeedType, SourceRepository> feedToSource, 
+            ILogger<RegistrationIdValidator> logger) 
+            : base(feedToSource, logger)
+        {
+        }
+
+        public override async Task CompareIndex(ValidationContext data, PackageRegistrationIndexMetadata v2, PackageRegistrationIndexMetadata v3)
```

Comment:
Probably would be worth to make a helper method to share the `Compare*` implementation between this and other classes below.
Something like
```
public static void AreEqual<T>(PackageRegistrationIndexMetadata v2, PackageRegistrationIndexMetadata v3, Func<PackageRegistrationIndexMetadata, T> extractValue, string nameOfValue)
{
    if (v2 == null || v3 == null) 
    {
        return; 
    } 
    var isEqual = extractValue(v2) == extractValue(v3);
    if (!isEqual) 
    { 
        throw new MetadataInconsistencyException<PackageRegistrationIndexMetadata>(v2, v3, 
            $"{nameOfValue} does not match!"); 
    } 
}
```